kvbc/tools/db_integrity_check/integrity_checker.cpp

```diff
@@ -219,15 +223,22 @@ concord::kvbc::categorization::RawBlock IntegrityChecker::getBlock(const BlockId
   return kvbc::categorization::RawBlock::deserialize(rawBlockSer);
 }
 
-void IntegrityChecker::validateAll() const {
+Digest IntegrityChecker::computeBlockDigest(const BlockId& block_id, const std::string_view& block) const {
+  Digest calcDigest;
+  BCStateTran::computeDigestOfBlock(block_id, block.data(), block.size(), &calcDigest);
+
+  return calcDigest;
+}
+
+void IntegrityChecker::validateRange(BlockId until_block) const {
   auto [block_id, digest] = getLatestsCheckpointDescriptor();
-  for (auto block = block_id; block > 0; --block) digest = checkBlock(block, digest);
+  for (auto block = block_id; block > until_block; --block) digest = checkBlock(block, digest);
```

Comment:
Maybe rename to last_block? 
(It's not that important, until_block just sounds weird)